pkg/transform/kubernetes/populators.go

```diff
@@ -0,0 +1,158 @@
+package kubernetes
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/kedgeproject/kedge/pkg/spec"
+
+	"sort"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	api_v1 "k8s.io/client-go/pkg/api/v1"
+)
+
+func populateProbes(c spec.Container) (spec.Container, error) {
+	// check if health and liveness given together
+	if c.Health != nil && (c.ReadinessProbe != nil || c.LivenessProbe != nil) {
+		return c, fmt.Errorf("cannot define field 'health' and " +
+			"'livnessProbe' or 'readinessProbe' together")
+	}
+	if c.Health != nil {
+		c.LivenessProbe = c.Health
+		c.ReadinessProbe = c.Health
+		c.Health = nil
+	}
+	return c, nil
+}
+
+func searchConfigMap(cms []spec.ConfigMapMod, name string) (spec.ConfigMapMod, error) {
+	for _, cm := range cms {
+		if cm.Name == name {
+			return cm, nil
+		}
+	}
+	return spec.ConfigMapMod{}, fmt.Errorf("configMap %q not found", name)
+}
+
+func getSecretDataKeys(secrets []spec.SecretMod, name string) ([]string, error) {
+	var dataKeys []string
+	for _, secret := range secrets {
+		if secret.Name == name {
+			for dk := range secret.Data {
+				dataKeys = append(dataKeys, dk)
+			}
+			for sdk := range secret.StringData {
+				dataKeys = append(dataKeys, sdk)
+			}
+			return dataKeys, nil
+		}
+	}
+	return nil, fmt.Errorf("secret %q not found", name)
+}
+
+func convertMapToList(m map[string]string) []string {
+	var d []string
+	for k := range m {
+		d = append(d, k)
+	}
+	return d
+}
+
+func convertEnvFromToEnvs(envFrom []api_v1.EnvFromSource, cms []spec.ConfigMapMod, secrets []spec.SecretMod) ([]api_v1.EnvVar, error) {
+	var envs []api_v1.EnvVar
+
+	// we will iterate on all envFroms
+	for ei, e := range envFrom {
+		if e.ConfigMapRef != nil {
+
+			cmName := e.ConfigMapRef.Name
+
+			// see if the configMap name which is given actually exists
+			cm, err := searchConfigMap(cms, cmName)
+			if err != nil {
+				return nil, errors.Wrapf(err, "envFrom[%d].configMapRef.name", ei)
+			}
+			// once that configMap is found extract all data from it and create a env out of it
+			configMapKeys := convertMapToList(cm.Data)
+			sort.Strings(configMapKeys)
```

Comment:
do we need to sort here? If we are doing this only for the tests, then why not sort in the tests instead of sorting here?
Also, a comment would help explaining why are we sorting this data?